dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2639,325 @@ void FileSprayer::pushParts()
     performTransfer();
 }
 
+using namespace Azure::Storage;
+using namespace Azure::Storage::Files;
+using namespace Azure::Storage::Blobs;
+
+enum ApiCopyStatus { Pending, Success, Failed, Aborted };
+interface IAPICopyClient : implements IInterface
+{
+    virtual ApiCopyStatus startCopy(const char *target, const char * source) = 0;
+    virtual ApiCopyStatus getProgress(CDateTime & dateTime, int64_t & outputLength) = 0;
+    virtual ApiCopyStatus abortCopy() = 0;
+    virtual ApiCopyStatus getStatus() const = 0;
+};
+
+class CApiCopyClient : implements IAPICopyClient, public CInterface
+{
+protected:
+    ApiCopyStatus status;
+public:
+    IMPLEMENT_IINTERFACE
+    virtual ApiCopyStatus getStatus() const
+    {
+        return status;
+    }
+    static bool buildAzureStorageURIBase(IStoragePlane * storagePlane, StringBuffer & uri)
+    {
+        const char *accountName = storagePlane->queryStorageApiAccount();
+        if (isEmptyString(accountName))
+        {
+            UWARNLOG("Unable to API copy: no account name provided for storage plane");
+            return false;
+        }
+        uri.appendf("https://%s", accountName);
+
+        if (storagePlane->getStorageType()==StorageType::StorageTypeAzureFile)
+        {
+            uri.append(".file");
+        }
+        else
+        {
+            dbgassertex(storagePlane->getStorageType()==StorageType::StorageTypeAzureBlob);
+            uri.append(".blob");
+        }
+        const char *container = storagePlane->queryStorageContainer();
+        if (isEmptyString(container))
+        {
+            UWARNLOG("Unable to API copy: no container name provided for storage plane");
+            return false;
+        }
+        uri.appendf(".core.windows.net/%s", container);
+
+        return true;
+    }
+    static void buildAzureStorageURI(IDistributedFile *distFile, unsigned partNum, const char * baseURI, const char *sasToken, StringBuffer & uri)
+    {
+        StringBuffer path;
+        bool ret = distFile->getUnmountedPath(partNum, path);
+        assertex(ret);
+        StringBuffer tmp;
+        uri.appendf("%s/%s%s", baseURI, encode_url(tmp, path.str()).str(), sasToken);
+    }
+};
+
+class AzureFileClient : public CApiCopyClient
+{
+    std::unique_ptr<Shares::ShareFileClient> fileClient;
+    std::unique_ptr<Shares::StartFileCopyOperation> fileCopyOp;
+public:
+    ApiCopyStatus startCopy(const char *target, const char * source)
+    {
+        try
+        {
+            fileClient.reset(new Shares::ShareFileClient(target));
+            fileCopyOp = std::make_unique<Shares::StartFileCopyOperation>(std::move(fileClient->StartCopy(source)));
+        }
+        catch (const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("AzureFileClient startCopy failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            status = ApiCopyStatus::Failed;
```

Comment:
the status will be overwritten by the following code to set it to pending.  Missing a return?

Reply:
status has been moved within the try block so it is not .